src/Attestation/Attestation/Models/AttestationClient.cs

```diff
@@ -0,0 +1,122 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+// TODO: Remove IfDef
+#if NETSTANDARD
+using Microsoft.Azure.Graph.RBAC.Version1_6.ActiveDirectory;
+#else
+using Microsoft.Azure.ActiveDirectory.GraphClient;
+#endif
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+using System.Net;
+using Microsoft.Azure.Commands.Common.Authentication;
+using Microsoft.Azure.Commands.ResourceManager.Common.Tags;
+using Microsoft.Azure.Management.Attestation;
+using PSAttestationProperties = Microsoft.Azure.Commands.Attestation.Properties;
+using Microsoft.Azure.Management.Attestation.Models;
+using Microsoft.Azure.Commands.Common.Authentication.Abstractions;
+using Microsoft.Rest.Azure;
+using Microsoft.Azure.Commands.Attestation.Models;
+using Newtonsoft.Json;
+using ErrorResponse = Microsoft.Azure.Management.Monitor.Version2018_09_01.Models.ErrorResponse;
+
+namespace Microsoft.Azure.Commands.Attestation.Models
+{
+    public class AttestationClient
+    {
+        private readonly AttestationManagementClient attestationClient;
+
+        public AttestationClient(IAzureContext context)
+        {
+            _subscriptionId = context.Subscription.GetId();
+            attestationClient = AzureSession.Instance.ClientFactory.CreateArmClient<AttestationManagementClient>(context, AzureEnvironment.Endpoint.ResourceManager);
+        }
+
+        public AttestationClient()
+        {
+        }
+        private readonly Guid _subscriptionId;
+        public PSAttestation CreateNewAttestation(AttestationCreationParameters parameters)
+        {
+            if (parameters == null)
+                throw new ArgumentNullException("parameters");
+            if (string.IsNullOrWhiteSpace(parameters.ProviderName))
+                throw new ArgumentNullException("parameters.ProviderName");
+            if (string.IsNullOrWhiteSpace(parameters.ResourceGroupName))
+                throw new ArgumentNullException("parameters.ResourceGroupName");
+            try
+            {
+                AttestationServiceCreationParams _creationParams = new AttestationServiceCreationParams();
+                if (!string.IsNullOrEmpty(parameters.AttestationPolicy))
+                {
+                    _creationParams.AttestationPolicy = parameters.AttestationPolicy;
+                }
+
+                var response = attestationClient.AttestationProviders.Create(
+                    resourceGroupName: parameters.ResourceGroupName,
+                    providerName: parameters.ProviderName,
+                    creationParams: _creationParams);
+
+                if (response.Status == "Ready" || response.Status == "NotReady")
```

Comment:
@shleiAmy nit: please make sure there are curly braces for if-statements

```cs
if (response.Status == "Ready" || response.Status == "NotReady")
{
    return new PSAttestation(response);
}
else
{
    return null;
}
```

This comment applies everywhere in this PR.

Reply:
changed